File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/json/Token.java
package org.kevoree.modeling.api.json;

/**
 * Created by duke on 10/3/14.
 */
public class Token {
    private Type tokenType;
    private Object value;

    public Token(Type tokenType, Object value) {
        this.tokenType = tokenType;
        this.value = value;
    }

    @Override
    public String toString() {
        String v;
        if (value != null) {
            v = " (" + value + ")";
        } else {
            v = "";
        }
        String result = tokenType.toString() + v;
        return result;
    }


    public Type getTokenType() {
        return tokenType;
    }

    public Object getValue() {
        return value;
    }
}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/json/Token.ts

class Token {

  private _tokenType: Type = null;
  private _value: any = null;

  constructor(p_tokenType: Type, p_value: any) {
    this._tokenType = p_tokenType;
    this._value = p_value;
  }

  public toString(): string {
    var v: string;
    if (this._value != null) {
      v = " (" + this._value + ")";
    } else {
      v = "";
    }
    var result: string = this._tokenType.toString() + v;
    return result;
  }

  public tokenType(): Type {
    return this._tokenType;
  }

  public value(): any {
    return this._value;
  }

}

